components/dashboards-web-component/src/viewer/DashboardViewPage.jsx

```diff
@@ -272,6 +418,234 @@ class DashboardViewPage extends Component {
+                if (headerImg != null) {
+                    pdf.addImage(headerImg, "JPEG", pdfConfig.stampImageDashboard.coordinates.x,
+                                 pdfConfig.stampImageDashboard.coordinates.y, pdfConfig.stampImageDashboard.size.x,
+                                 pdfConfig.stampImageDashboard.size.y);
+                }
+            }
+        });
+
+        clearInterval(this.timer);
+        this.handleReportStatusClose();
+        this.state.completed = 0;
+        this.state.includeTime = false;
+
+        const docName = title + ".pdf";
+        pdf.save(docName);
+
+    }
+
+    // Get the image from deployment.yaml file through the server
+    async addPdfImage(url, callback) {
+        let path = "/portal/public/app/images/"
+
+        await axios.get(url, { auth: { username: "admin", password: "admin" } })
+            .then(function (res) {
+                if (res.data === '') {
+                    callback('none');
+                } else {
+                    path += res.data;
+
+                    //Convert image to bdase64 encoded image (data_url)
+                    const canvas = document.createElement('canvas');
+                    const ctx = canvas.getContext('2d');
+                    const img = new Image();
+                    let imgStr;
+
+                    img.onload = function () {
+                        canvas.width = img.width;
+                        canvas.height = img.height;
+                        ctx.drawImage(img, 0, 0);
+                        imgStr = canvas.toDataURL("image/png");
+                        callback(imgStr);
+                    }
+
+                    img.src = path;
+                }
+
+            });
+    }
+
+    async removePage(index) {
+
+        const newAr = this.state.pageList.slice();
+        const ind = newAr.indexOf(index);
+        newAr.splice(ind, 1);
+        this.setState({ pageList: newAr });
+
+        if (newAr.indexOf(index) == -1) {
+            localStorage.removeItem(index);
+        }
+
+    }
+
+    async capturePage(index) {
+
+        const url = window.location.href;
+        const parts = url.split('/');
+        const currentPage = parts[parts.length - 1] === "" ? parts[parts.length - 2] : parts[parts.length - 1];
+
+        await html2cavas(document.getElementById("dashboard-container")).then((canvas) => {
+            const imgData = canvas.toDataURL('image/png');
+            const w = canvas.width;
+            const h = canvas.height;
+            localStorage.setItem(currentPage, JSON.stringify({ imageData: imgData, width: w, height: h }));
+
+            const newAr = this.state.pageList.slice();
+            newAr.push(currentPage);
+            this.setState({ pageList: newAr });
+
+            const val = 620 / canvas.width;
+            this.state.resizeRatio = val;
+            this.state.canvasWidth = canvas.width;
+            this.state.canvasHeight = canvas.height;
+        });
+    }
+
```

Comment:
DashboardViewPage.jsx is too long. Let's try to break into Utill classes

Reply:
Done